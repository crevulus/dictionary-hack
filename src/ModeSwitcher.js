import React, { Component } from 'react';
import Dictionary from './Dictionary';
import Switch from 'react-switch'
import Thesaurus from './Thesaurus';

class ModeSwitcher extends Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: false
    }
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(checked) {
    this.setState({checked});
  }

  render() {
    return (
      <div>
        <p className="thesaurus-text">Thesaurus Mode: <Switch
          className="switch"
          onChange={this.handleChange}
          checked={this.state.checked}
          /></p>
        {this.state.checked ? <Thesaurus /> : <Dictionary />}
      </div>
    );
  }
}

export default ModeSwitcher;
