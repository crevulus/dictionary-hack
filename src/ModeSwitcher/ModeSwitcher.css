.thesaurus-text {
  font-weight: 800;
  color: white;
}