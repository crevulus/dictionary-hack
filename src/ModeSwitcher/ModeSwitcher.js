import React, { Component } from 'react';
import Dictionary from '../Dictionary/Dictionary';
import Switch from 'react-switch'
import Thesaurus from '../Thesaurus/Thesaurus';
import './ModeSwitcher.css'

class ModeSwitcher extends Component {
  constructor(props) {
    super(props);
    this.state = {
      checked: false
    }
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(checked) {
    this.setState({checked});
  }

  render() {
    return (
      <div>
        <p className="thesaurus-text">Thesaurus Mode: <Switch
          className="switch"
          onChange={this.handleChange}
          checked={this.state.checked}
          onColor="#000"
          /></p>
        {this.state.checked ? <Thesaurus /> : <Dictionary />}
      </div>
    );
  }
}

export default ModeSwitcher;
