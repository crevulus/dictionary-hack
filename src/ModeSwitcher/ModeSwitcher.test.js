/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import ModeSwitcher from './ModeSwitcher';
import renderer from 'react-test-renderer';

test('renders component', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ModeSwitcher></ModeSwitcher>, div);
});

test('matches snapshot', () => {
  const snap = renderer.create(<ModeSwitcher></ModeSwitcher>).toJSON();
  expect(snap).toMatchSnapshot();
});