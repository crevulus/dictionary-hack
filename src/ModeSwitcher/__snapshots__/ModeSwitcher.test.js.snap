// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <p
    className="thesaurus-text"
  >
    Thesaurus Mode: 
    <div
      className="switch"
      style={
        Object {
          "MozTransition": "opacity 0.25s",
          "MozUserSelect": "none",
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "WebkitTransition": "opacity 0.25s",
          "WebkitUserSelect": "none",
          "borderRadius": 14,
          "direction": "ltr",
          "display": "inline-block",
          "msUserSelect": "none",
          "opacity": 1,
          "position": "relative",
          "textAlign": "left",
          "touchAction": "none",
          "transition": "opacity 0.25s",
          "userSelect": "none",
        }
      }
    >
      <div
        className="react-switch-bg"
        onClick={[Function]}
        onMouseDown={[Function]}
        style={
          Object {
            "MozTransition": "background 0.25s",
            "WebkitTransition": "background 0.25s",
            "background": "#888888",
            "borderRadius": 14,
            "cursor": "pointer",
            "height": 28,
            "margin": 0,
            "position": "relative",
            "transition": "background 0.25s",
            "width": 56,
          }
        }
      >
        <div
          style={
            Object {
              "MozTransition": "opacity 0.25s",
              "WebkitTransition": "opacity 0.25s",
              "height": 28,
              "opacity": 0,
              "pointerEvents": "none",
              "position": "relative",
              "transition": "opacity 0.25s",
              "width": 30,
            }
          }
        >
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 17 21"
            width="100%"
          >
            <path
              d="M11.264 0L5.26 6.004 2.103 2.847 0 4.95l5.26 5.26 8.108-8.107L11.264 0"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <div
          style={
            Object {
              "MozTransition": "opacity 0.25s",
              "WebkitTransition": "opacity 0.25s",
              "height": 28,
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "right": 0,
              "top": 0,
              "transition": "opacity 0.25s",
              "width": 30,
            }
          }
        >
          <svg
            height="100%"
            style={
              Object {
                "position": "absolute",
                "top": 0,
              }
            }
            viewBox="-2 -5 14 20"
            width="100%"
          >
            <path
              d="M9.9 2.12L7.78 0 4.95 2.828 2.12 0 0 2.12l2.83 2.83L0 7.776 2.123 9.9 4.95 7.07 7.78 9.9 9.9 7.776 7.072 4.95 9.9 2.12"
              fill="#fff"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </div>
      <div
        className="react-switch-handle"
        onClick={[Function]}
        onMouseDown={[Function]}
        onTouchCancel={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "MozTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
            "WebkitTransition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
            "background": "#ffffff",
            "border": 0,
            "borderRadius": "50%",
            "boxShadow": null,
            "cursor": "pointer",
            "display": "inline-block",
            "height": 26,
            "outline": 0,
            "position": "absolute",
            "top": 1,
            "transform": "translateX(1px)",
            "transition": "background-color 0.25s, transform 0.25s, box-shadow 0.15s",
            "width": 26,
          }
        }
      />
      <input
        checked={false}
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyUp={[Function]}
        role="switch"
        style={
          Object {
            "border": 0,
            "clip": "rect(0 0 0 0)",
            "height": 1,
            "margin": -1,
            "overflow": "hidden",
            "padding": 0,
            "position": "absolute",
            "width": 1,
          }
        }
        type="checkbox"
      />
    </div>
  </p>
  <div
    className="dictionary-div"
  >
    <form
      onSubmit={[Function]}
    >
      <label
        className="search-text"
      >
        Search for a word:
        <br />
        <input
          label="Search"
          onChange={[Function]}
          type="text"
        />
      </label>
      <input
        className="submit"
        onClick={[Function]}
        type="submit"
        value="Search"
      />
    </form>
    <div
      className="data"
    >
      <p
        className="entry"
      >
        Search for something!
      </p>
      <p
        className="word"
      >
        
      </p>
      <p
        className="pronunciation"
      />
      <p
        className="category"
      />
      <p
        className="definition"
      />
      <p
        className="synonyms"
      />
    </div>
  </div>
</div>
`;
